opentelemetry-sdk/src/logs/record.rs

```diff
@@ -201,6 +202,73 @@ impl SdkLogRecord {
             .flatten()
             .any(|(k, v)| k == key && v == value)
     }
+
+    /// Updates the first occurrence of an attribute with the specified key.
+    ///
+    /// This method searches for the first occurrence of the attribute with the given key
+    /// in the `attributes` collection. If the key is found, its value is updated with the
+    /// provided value. If the key is not found, the attribute is added.
```

Comment:
> If the key is not found, the attribute is added.

We already have an `add_attribute` method which is meant for adding a new attribute. I think we should do one of the following:
1. Either make `update_attribute` strictly an update operation.
2. Or rename the method to `add_or_update_attribute` to reflect its true behavior.